### PR TITLE
Build with .NET 8

### PR DESCRIPTION
**Pull request by borland**
**_Are you a customer of Octopus Deploy? Please contact [our support team](https://octopus.com/support) so we can triage your PR, so that we can make sure it's handled appropriately._**

# Background

# Results

Fixes https://github.com/OctopusDeploy/Issues/issues/... _(optional public issue)_

Fixes https://github.com/OctopusDeploy/ResearchAndDevelopment/issues/... _(optional private issue)_

See [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) (including [this flowchart](https://whimsical.com/r-d-incoming-work-workflow-aug-21-NsDnGQXcwBLwU66a88Zhue) 

## Before

Before

## After

After

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: build/Build.cs [hunk 1]
```diff
@@ -108,7 +108,7 @@ ITargetDefinition CompileDefinition(ITargetDefinition targetDefinition, [CanBeNu
     Target TestWindowsNet48 => _ => TestDefinition(_, CompileNet48, "net48", runDotMemoryTests: true);
 
     [PublicAPI]
-    Target TestWindowsNet60 => _ => TestDefinition(_, CompileNet60, "net6.0", runDotMemoryTests: true);
+    Target TestWindowsNet80 => _ => TestDefinition(_, CompileNet80, "net8.0", runDotMemoryTests: true);
```
Comment (nathanwoctopusdeploy) [inline build/Build.cs:111]: I think we would want to keep build net6.0 and test net6.0 and then add the net8.0 versions.

File: source/Halibut.Tests/Halibut.Tests.csproj [hunk 2]
```diff
@@ -51,10 +51,11 @@
     <PackageReference Include="Serilog.Sinks.ColoredConsole" Version="3.0.1" />
   </ItemGroup>
 
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'net48' ">
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net4.8' ">
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.Tests/Halibut.Tests.csproj:54]: Is this correct?

Comment (borland) [reply source/Halibut.Tests/Halibut.Tests.csproj:54]: It's wrong, it should be "net48". Perils of putting down a PR for a week or two then picking it back up again, sorry

File: source/Octopus.TestPortForwarder/Octopus.TestPortForwarder.csproj [hunk 3]
```diff
@@ -8,10 +8,11 @@
     </PropertyGroup>
 
     <PropertyGroup Condition="!$([MSBuild]::IsOSUnixLike())">
+        <TargetFrameworks>net48;net6.0;net8.0</TargetFrameworks>
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.TestPortForwarder/Octopus.TestPortForwarder.csproj:11]: Double up

File: source/Halibut.Tests/Halibut.Tests.csproj [hunk 4]
```diff
@@ -51,10 +51,11 @@
...
     <DefineConstants>$(DefineConstants);SUPPORTS_WEB_SOCKET_CLIENT;DOES_NOT_SUPPORT_CANCELLATION_ON_SOCKETS</DefineConstants>
   </PropertyGroup>
 
+    <PropertyGroup Condition=" '$(TargetFramework)' == 'net6.0' Or '$(TargetFramework)' == 'net8.0' ">
```
[the remark this answers is not in the document]
Comment (borland) [reply source/Halibut.Tests/Halibut.Tests.csproj:58]: The general plan we've been following is:

- Switch the compilers to use .NET 8
- Build the library only targeting .NET 6 unless there's some special reason not to
- Have the tests run on both .NET6 and .NET8

But Halibut is complicated and messy, let's have a chat about the strategy for it

File: source/Octopus.TestPortForwarder/Octopus.TestPortForwarder.csproj (hunk 3 above)
Comment (borland) [reply source/Octopus.TestPortForwarder/Octopus.TestPortForwarder.csproj:11]: net6.0 and net8.0 aren't double-ups?

Comment (nathanwoctopusdeploy) [reply source/Octopus.TestPortForwarder/Octopus.TestPortForwarder.csproj:11]: image

File: source/Halibut.Tests/Halibut.Tests.csproj (hunk 4 above)
[the remark this answers is not in the document]
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Halibut.Tests.csproj:58]: image

**Pull request closed without merging**